plugins/providers/grafana/client_test.go

```diff
@@ -32,5 +42,474 @@ func TestNewClient(t *testing.T) {
+		s.mockHttpClient.On("Do", testRequest).Return(&dashboardResponse, nil).Once()
+
+		result, err1 := s.client.GetDashboards(folderID)
+		var dashboards []grafana.Dashboard
+		for _, db := range result {
+			dashboards = append(dashboards, *db)
+		}
+
+		s.Nil(err1)
+		s.Equal(expectedDashboards, dashboards)
+	})
+}
+
+func (s *ClientTestSuite) getTestRequest(method, path string, body interface{}) (*http.Request, error) {
+	basicKey := "dGVzdC11c2VybmFtZTp0ZXN0LXBhc3N3b3Jk"
+	var buf io.ReadWriter
+	if body != nil {
+		buf = new(bytes.Buffer)
+		err := json.NewEncoder(buf).Encode(body)
+		if err != nil {
+			return nil, err
+		}
+	}
+	url := fmt.Sprintf("%s%s", s.host, path)
+	req, err := http.NewRequest(method, url, buf)
+	if err != nil {
+		return nil, err
+	}
+	if body != nil {
+		req.Header.Set("Content-Type", "application/json")
+	}
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Authorization", "Basic "+basicKey)
+	req.Header.Set("X-Grafana-Org-Id", s.org)
+
+	return req, nil
+}
+
+func (s *ClientTestSuite) TestGrantDashboardAccess() {
+
+	s.Run("should return error if user not found", func() {
+		s.setup()
+
+		user := "test-email@gojek.com"
+		role := "view"
+		expectedError := errors.New("Error in getting user")
+		url := fmt.Sprintf("/api/users/lookup?loginOrEmail=%s", user) //testing the getUser(user) Response
+		testRequest, err := s.getTestRequest(http.MethodGet, url, nil)
+		s.Require().NoError(err)
+
+		userResponse := http.Response{StatusCode: 404, Body: ioutil.NopCloser(bytes.NewReader([]byte(nil)))}
+		s.mockHttpClient.On("Do", testRequest).Return(&userResponse, expectedError).Once()
+
+		resource := grafana.Dashboard{
+			ID:          1,
+			Title:       "db_1",
+			FolderID:    50,
+			FolderTitle: "fd_1",
+		}
+
+		actualError := s.client.GrantDashboardAccess(&resource, user, role)
+		s.Equal(expectedError, actualError)
+	})
+
+	s.Run("should return error if user not found", func() {
+		s.setup()
+
+		user := "test-email@gojek.com"
+		role := "view"
+
+		url := fmt.Sprintf("/api/users/lookup?loginOrEmail=%s", user) //testing the getUser(user) Response
+		testRequest, err := s.getTestRequest(http.MethodGet, url, nil)
+		s.Require().NoError(err)
+
+		userResponseJSON := `{ "id":55,"email":"test-email@gojek.com" }`
+		userResponse := http.Response{StatusCode: 404, Body: ioutil.NopCloser(bytes.NewReader([]byte(userResponseJSON)))}
+		s.mockHttpClient.On("Do", testRequest).Return(&userResponse, nil).Once()
+
+		resource := grafana.Dashboard{
+			ID:          1,
+			Title:       "db_1",
+			FolderID:    50,
+			FolderTitle: "fd_1",
+		}
+
+		actualError := s.client.GrantDashboardAccess(&resource, user, role)
+		e := grafana.ErrUserNotFound
+		s.Equal(e, actualError)
+	})
+
+	s.Run("should return an error if permissions are invalid for dashboard access", func() {
+		s.setup()
+
+		user := "test-email@gojek.com"
+		role := "invalid role" //valid roles are "view", "edit", "admin"
```

Comment:
```suggestion
		role := "invalid role"
```
I don't think we need to add this comment here